pubsub.go

```diff
@@ -187,18 +189,58 @@ func NewPubSub(ctx context.Context, h host.Host, rt PubSubRouter, opts ...Option
 	return ps, nil
 }
 
+// WithValidateThrottle sets the upper bound on the number of active validation
+// goroutines.
 func WithValidateThrottle(n int) Option {
 	return func(ps *PubSub) error {
 		ps.validateThrottle = make(chan struct{}, n)
 		return nil
 	}
 }
 
-func WithMessageSigning(strict bool) Option {
+// WithMessageSigning enables or disables message signing (enabled by default).
+func WithMessageSigning(enabled bool) Option {
 	return func(p *PubSub) error {
-		p.signID = p.host.ID()
-		p.signKey = p.host.Peerstore().PrivKey(p.signID)
-		p.signStrict = strict
+		if enabled {
+			p.signKey = p.host.Peerstore().PrivKey(p.signID)
+			if p.signKey == nil {
+				return fmt.Errorf("can't sign for peer %s: no private key", p.signID)
+			}
```

Comment:
Shouldn't we set the signID as well for consistency?
Also, how well does it combine with the `WithMessageAuthor` option? Seems that this need to come first.

Reply:
Either way should work.

* WithMessageSigning just uses the existing signID (defaults to the host ID).
* WithMessageAuthor *sets* the signID.

* If you pass WithMessgaeSigning first, WithMessageAuthor will update the signKey.
* If you pass WithMessageAuthor first, WIthMessageSigning will use that author.